lib/editing/editor.dart

```diff
@@ -186,10 +186,22 @@ class Completion {
   /// This can be `null`.
   final int cursorOffset;
 
+  /// Quick fixes that alter multiple lines of editor content can't simply
+  /// provide an offset on the cursor's current line to represent the position
+  /// to which the cursor should move. This field allows for a character count
+  /// from the top of the file to be used instead.
+  final int absoluteCursorPosition;
```

Comment:
Ideally this would be a `List<int>` and `_completionHelper` be responsible for either using the first or all of the cursor positions

Reply:
Yeah, there are a few forms in which cursor info can be created by the analyzer:

* Completions have a relative cursor offset, so that some have the cursor start immediately afterward, and method name completions have it start in between the parens. These could likely be converted to use the absolute one.
* A single, absolute cursor position is used for a one-shot fix like "Move Widget Down."
* A series of post-application "linked edit groups" require repeated multi-cursor, multi-selection editing by the user.

The first of those already worked when we showed up to the project. The second maps nicely to a single, absolute position, which is what I've done here. The third is much more complicated and is largely out of scope for this particular issue. As a stopgap, I'm translating the lined edit group info provided by the analyzer into a single absolute cursor position so we don't give the user something completely useless, but (as you point out), it's not ideal and will need to be revisited as our polishing efforts continue.

Reply:
> The third is much more complicated and is largely out of scope for this particular issue

I agree, I was just suggesting we make the type a List<int> so that #3 goes more smoothly when we get to it.